firmware/main.c

```diff
@@ -692,131 +689,128 @@ static void cmd_motor_rotary(BaseSequentialStream *chp, int argc, char *argv[])
+				sprintf(direction_literal, "clockwise (CW)");
+			else if(direction == CCW)
+				sprintf(direction_literal, "counterclockwise (CCW)");
+			// Each sent pulse will result in 1.8 degrees rotation
+			rotational_angle = pulses * 1.8;
+			chprintf(chp, "Number of pulses sent: %d\r\nDirection: %s\r\n", geared_pulses, direction_literal);
+			chprintf(chp, "Angle of rotation: %.1f\r\n", rotational_angle);
+			// reset direction pin at the end of the operation to 0
+			palClearPad(pinPorts[pinDirection].gpio, pinPorts[pinDirection].pin);
 			break;
-
-		case 3: // Turn 45 degree CCW
-			// !fall through!
-		case 4: // Homing Mode
-			// !fall through from above!
+		// 2: home
+		case 2:
+			chprintf(chp, "Selected mode: home\r\n");
+			chprintf(chp, "\tRotary unit returns to reference position\r\n");
 			palSetPad(pinPorts[pinEnable].gpio, pinPorts[pinEnable].pin);
 			chThdSleepMilliseconds(200);
 			palClearPad(pinPorts[pinEnable].gpio, pinPorts[pinEnable].pin);
 			break;
-
 		default:
 			chprintf(chp, "Select one of the available modes\r\n");
 			break;
 	}
 	return;
 
+
 exit_with_usage:
-	chprintf(chp, "Usage: motor_rotary -m <mode> -p [pulses] -d [direction]\r\n"
-			"\tNumber of pulses to turn (1 pulse = 1,8 degrees the motor and 0,0375 degrees the rotary table)\r\n"
+	chprintf(chp, "Usage: motor_rotary <mode> <arguments>\r\n"
+			"\tNumber of pulses to turn "
+			"(1 pulse = 1,8 degrees the motor and 0,0375 degrees the rotary table)\r\n"
+			"\tNote: The given number of pulses is multiplied by 48 (gear reduction).\r\n"
+			"\t      -> 1 pulse as argument sends 48 pulses to the motor and "
+			"turns the rotary table by 1.8 degrees.\r\n"
 			"\tModes:\r\n"
-			"\t  1 - Clock mode, turn left --> [require -p (number of pulses) and -d (direction of rotation)]\r\n"
-			"\t  2 - Clock mode, turn right --> [require -p (number of pulses) and -d (direction of rotation)]\r\n"
-			"\t  3 - Turn 45 degree CW\r\n"
-			"\t  4 - Ref. run with external sensor (Homing)\r\n"
-			"\tDirection:\r\n"
-			"\t  0 - CW\r\n"
-			"\t  1 - CCW\r\n");
+			"\t 'home' - move to reference position to calibrate angle measurement\r\n"
+			"\t 'rotate' <pulses>\r\n"
+			"\t\t <pulses> Option 1: +(number of pulses as int) - clockwise (CW) rotation\r\n"
+			"\t\t <pulses> Option 2: -(number of pulses as int) - counterclockwise (CCW) rotation\r\n"
+			);
 }
 
 
 static void cmd_motor_linear(BaseSequentialStream *chp, int argc, char *argv[]) {
+	// Pin declaration for use with the C5-E (gpio number arbitrary)
 	const char maplePinPulses[] = "26";
 	const char maplePinDirection[] = "28";
 	const char maplePinEnable[] = "22";
 
 	int pinPulses, pinDirection, pinEnable;
 
-	char * paramMode = NULL;
-	int mode = -1;
+	// available modes
+	const int MOVE = 1, HOME = 2;
 
-	char * paramDirection = NULL;
+	int mode = -1;
 	int direction = -1;
-
-	char * paramPulses = NULL;
 	int pulses = -1;
 
+	// define motion directions: towards switchboard/wall
+	int SWITCHBOARD = 0, WALL = 1;
```

Comment:
enum.
also, those names are a bit arbitrary.
maybe use the motor as a reference, not some random setup?